### PR TITLE
Update workshop for prometheus python client

### PR DESCRIPTION
**Pull request by sleepypioneer**
- updates a few files like docker-compose and requirements which I realised needed fixing when I ran it.
- Updates Readme for the python library.

I got this working, pretty painlessly but following how I had done it previously not the demo example on their site. I find this much cleaner. We do get metrics out of the box after adding the handler so that is cool. Perhaps we should know a bit better what those metrics are.

Maybe we want to add at the end another step or "homework" to add different types of metrics, we could talk about them in the slides for example.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+install:
+	cd ./app && pipenv install
+
+dev: install
+	cd ./app && pipenv run python main.py
+
```
Comment (emilywoods) [inline Makefile]: some extra lines :)

File: README.md [hunk 2]
```diff
@@ -7,3 +7,83 @@ In the directory `app/`, we have a simple Python application. We want to start o
 We will do this using the time-series database system [Prometheus](https://prometheus.io), which uses a "pull" method to extract data from running applications. This means that the applications need to "export" their data, so that Prometheus is able to "scrape" the metric data from them. This is typically done via an HTTP endpoint (`/metrics`, by convention).
 
 We will use the [Prometheus Python client library](https://github.com/prometheus/client_python) to track metrics in our code.
+
+## Content
+
+### Section 1: Exposing metrics
+
+For this section, you can use `make dev` to install depencies and run the dev server (you will need pipenv installed on your machine). (Docker will be used later)
```
Comment (emilywoods) [inline README.md]: might be useful to add a prerequisites section above with `pipenv`

Comment (emilywoods) [reply README.md]: Is the comment 

```
(Docker will be used later)
```
a note for the hosts or participants?

**Review by emilywoods — COMMENTED** [review]
nice work! i have left a couple of small comments

File: README.md (hunk 2 above)
Comment (sleepypioneer) [reply README.md]: I remember it as something we would say to the participants to make it clear we didn't use docker here but I would say its not so necessary and will remove it :)

File: README.md [hunk 3]
```diff
@@ -7,3 +7,89 @@ In the directory `app/`, we have a simple Python application. We want to start o
 We will do this using the time-series database system [Prometheus](https://prometheus.io), which uses a "pull" method to extract data from running applications. This means that the applications need to "export" their data, so that Prometheus is able to "scrape" the metric data from them. This is typically done via an HTTP endpoint (`/metrics`, by convention).
 
 We will use the [Prometheus Python client library](https://github.com/prometheus/client_python) to track metrics in our code.
+
+## Prerequisites
+
+For this workshop you will need [Pipenv](https://pipenv.pypa.io/en/latest/install/#installing-pipenv) and [Docker](https://docs.docker.com/get-docker/) running on your machine.
```
Comment (emilywoods) [inline README.md]: Should we also mention Python 3?

**Review by emilywoods — APPROVED** [review]
:umbrella:

File: README.md (hunk 3 above)
Comment (sleepypioneer) [inline README.md]:
```suggestion
For this workshop you will need [Python 3](https://installpython3.com/), [Pipenv](https://pipenv.pypa.io/en/latest/install/#installing-pipenv) and [Docker](https://docs.docker.com/get-docker/) running on your machine.
```

**Pull request merged**